integration-tests/tests/src/tests/sync/geospatial.ts

```diff
@@ -0,0 +1,920 @@
+type QueryExceptionPair = [ExpectedException: string, Query: string, ...QueryArgs: Array<any>];
+type QueryResultsPair = [ExpectedResults: any[], Query: string, ...QueryArgs: Array<any>];
+
+/**
+ * Teardown subscription helper, which will remove local data and assert the result
+ */
+
+async function tearDownSubscription(results: Realm.Results<any>, realm: Realm) {
+  results.unsubscribe();
+  await realm.syncSession?.downloadAllServerChanges();
+  expect(results.length).to.equal(0);
+}
+
+/**
+ * Helper method which runs an array of queries and asserts them to given expected length
+ * For example: (r, "Obj", [1, "intCol == 0"]) => querying "intCol == 0" should return 1 elements.
+ */
+const expectQueryLength = (realm: Realm, objectSchema: Realm.ObjectClass, queryLengthPairs: QueryLengthPair[]) => {
+  return Promise.all(
+    queryLengthPairs.map(async ([expectedLength, queryString, ...queryArgs]) => {
+      const filtered = realm.objects(objectSchema).filtered(queryString, ...queryArgs);
+
+      await filtered.subscribe({ behavior: WaitForSync.Always });
+      expect(filtered.length).to.equal(
+        expectedLength,
+        `Expected length ${expectedLength} for query: ${queryString} ${JSON.stringify(queryArgs)}`,
+      );
+
+      await tearDownSubscription(filtered, realm);
+    }),
+  );
+};
+
+/**
+ * Helper method which runs an array of queries and asserts them to throw given exception.
+ * For example: (r, "Obj", ["invalid", "intCol == #"]) => querying "intCol == #" should
+ * throw an exception which includes the string "invalid".
+ */
+const expectQueryException = async (
+  realm: Realm,
+  objectSchema: Realm.ObjectClass,
+  queryExceptionPairs: QueryExceptionPair[],
+) => {
+  return Promise.all(
+    queryExceptionPairs.map(async ([expectedException, queryString, ...queryArgs]) => {
+      expect(async () => {
+        const filtered = realm.objects(objectSchema).filtered(queryString, ...queryArgs);
+        await filtered.subscribe({ behavior: WaitForSync.Always });
+      }).throws(
+        expectedException,
+        `Expected exception not thrown for query: ${queryString} ${JSON.stringify(queryArgs)}`,
+      );
+    }),
+  );
+};
+
+/**
+ * Helper method which runs an array of queries and asserts the results' provided [propertyToCompare]
+ * field to equal the given expected result array's values.
+ * For example: (r, "Obj", "intCol" [[3, 4], "intCol > 2]) => querying "intCol > 2" should
+ * return results whose elements' intCol property values are 3 and 4.
+ */
+const expectQueryResultValues = async (
+  realm: Realm,
+  objectSchema: Realm.ObjectClass,
+  propertyToCompare: string,
+  queryResultPairs: QueryResultsPair[],
+) => {
+  return Promise.all(
+    queryResultPairs.map(async ([expectedResults, queryString, ...queryArgs]) => {
+      let results = realm.objects(objectSchema);
+      results = results.filtered(queryString, ...queryArgs);
```

Comment:
I just continued using `results` here but I guess it could also be `filtered` as in the other tests.
```suggestion
      const results = realm.objects(objectSchema).filtered(queryString, ...queryArgs);
```